src/Tribe/Aggregator/Record/Abstract.php

```diff
@@ -1128,59 +1138,131 @@ public function insert_posts( $items = array() ) {
 				unset( $event['feature_event'] );
 			}
 
-			if ( empty( $event['recurrence'] ) ) {
-				$non_recurring = true;
-			}
-
 			// set the parent
 			if ( ! empty( $event['ID'] ) && ( $id = wp_get_post_parent_id( $event['ID'] ) ) ) {
 				$event['post_parent'] = $id;
 			} elseif ( ! empty( $event['parent_uid'] ) && ( $k = array_search( $event['parent_uid'], $possible_parents ) ) ) {
 				$event['post_parent'] = $k;
 			}
 
-			//if we should create a venue or use existing
+			// if we should create a venue or use existing
 			if ( ! empty( $event['Venue']['Venue'] ) ) {
-				$v_id = array_search( $event['Venue']['Venue'], $found_venues );
-				if ( false !== $v_id ) {
-					$event['EventVenueID'] = $v_id;
-				} elseif ( $venue = get_page_by_title( $event['Venue']['Venue'], 'OBJECT', Tribe__Events__Main::VENUE_POST_TYPE ) ) {
-					$found_venues[ $venue->ID ] = $event['Venue']['Venue'];
-					$event['EventVenueID']      = $venue->ID;
-				} else {
-					$event['Venue']['ShowMap']     = $show_map_setting;
-					$event['Venue']['ShowMapLink'] = $show_map_setting;
-					$event['EventVenueID'] = Tribe__Events__Venue::instance()->create( $event['Venue'], $this->meta['post_status'] );
+				if ( ! empty( $item->venue->global_id ) ) {
+					// Did we find a Post with a matching Global ID in History
+					$venue = Tribe__Events__Aggregator__Event::get_post_by_meta( 'global_id_lineage', $item->venue->global_id );
+
+					var_dump( $venue );
+
+					// Save the Venue Data for Updating
+					$venue_data = $event['Venue'];
+
+					if ( $venue ) {
+						$event['EventVenueID'] = $venue->ID;
+						$found_venues[ $venue->ID ] = $event['Venue']['Venue'];
+
+						// Here we might need to update the Venue depending on the main GlobalID
+						// @todo Update Venue
```

Comment:
Just checking - does this `@todo` need to be resolved before merging, or are we good?